HerPortal.ManagementShell/CommandHandler.cs

```diff
@@ -0,0 +1,335 @@
+            }
+            else
+            {
+                outputProvider.Output("Add the following Local Authorities:");
+                PrintCodes(custodianCodes, code => custodianCodeToLaNameDict[code]);
+            }
+        });
+    }
+
+    private bool ConfirmRemoveCustodianCodes(string userEmailAddress, IReadOnlyCollection<string> custodianCodes)
+    {
+        return ConfirmChangesToDatabase(userEmailAddress, () =>
+        {
+            outputProvider.Output("Remove the following Local Authorities:");
+            PrintCodes(custodianCodes, code => custodianCodeToLaNameDict[code]);
+        });
+    }
+
+    private bool ConfirmAddConsortiumCodes(string? userEmailAddress, IReadOnlyCollection<string> consortiumCodes,
+        User? user)
+    {
+        return ConfirmChangesToDatabase(userEmailAddress, () =>
+        {
+            if (user != null)
+            {
+                outputProvider.Output("Add the following Consortia:");
+                PrintCodes(consortiumCodes, code => consortiumCodeToConsortiumNameDict[code]);
+
+                // flag the need to remove access for any LAs in the new consortia
+                var ownedCustodianCodesInConsortia =
+                    adminAction.GetOwnedCustodianCodesInConsortia(user, consortiumCodes);
+
+                outputProvider.Output("Remove the following Local Authorities in these Consortia:");
+                PrintCodes(ownedCustodianCodesInConsortia, code =>
+                    $"{custodianCodeToLaNameDict[code]} ({custodianCodeToConsortiumNameDict[code]})");
+            }
+            else
+            {
+                outputProvider.Output("Add the following Consortia:");
+                PrintCodes(consortiumCodes, code => consortiumCodeToConsortiumNameDict[code]);
+            }
+        });
+    }
+
+    private bool ConfirmRemoveConsortiumCodes(string? userEmailAddress, IReadOnlyCollection<string> consortiumCodes)
+    {
+        return ConfirmChangesToDatabase(userEmailAddress, () =>
+        {
+            outputProvider.Output("Remove the following Consortia:");
+            PrintCodes(consortiumCodes, code => consortiumCodeToConsortiumNameDict[code]);
+        });
+    }
+
+    private bool ConfirmChangesToDatabase(string? userEmailAddress, Action printAction)
+    {
+        outputProvider.Output(
+            $"You are changing permissions for user {userEmailAddress}:");
+
+        try
+        {
+            printAction();
+        }
+        catch (Exception e)
+        {
+            outputProvider.Output($"{e.Message} Process terminated");
+            return false;
+        }
+
+        var hasUserConfirmed = outputProvider.Confirm("Please confirm (y/n)");
+        if (!hasUserConfirmed) outputProvider.Output("Process cancelled, no changes were made to the database");
+
+        return hasUserConfirmed;
+    }
+
+    private void DisplayUserStatus(Enum status)
```

Comment:
Can we be more specific?
```suggestion
    private void DisplayUserStatus(AdminAction.UserStatus status)
```

(also would it make sense for `UserStatus` to live outside `AdminAction` now?)

Reply:
I think so too